### PR TITLE
Building vue.js app with django api and axios

### LINKED ISSUE #5401: Build Vue Application with Django REST API and Axios
#### Proposed title of article
Build Vue Application with Django REST API  and Axios

#### Proposed article introduction
When building web applications one may want to consume and display data from an API. This also applies to Vue.js applications where Django, a backend framework based on python is purely used as an API using a package called Django rest framework to build web applications together with Axios, a HTTP client library for fetching and displaying data.
To understand this, we will cover as much in this article to equip the reader with the understanding of how to combine Django and Vue.js to build web applications using Django rest framework together with Axios . The reader will understand how to configure rest API, Axios and their integration in building Vue web applications.
#### Key takeaways 
The reader will be able to:
- Understand how to build Django rest API using Django REST Framework.
- How to combine Django and Vue.js.
- How to build Vue application using Axios.

#### Article quality
This article will be of high quality different from the others, as it not only shows how to combine Django with Vue.js but also its implementation on how one can build Vue web applications using Django rest api together with Axios.

#### References
N/A

### Conclusion
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by dentonya**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5401

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by dentonya** [comment]
hey @mercymeave have done the necessary corrections regarding copy leaks in PR #5766, some of the code snippets cannot be changed since the syntax is same throughout.

**Comment by Kayere** [comment]
Hello @dentonya,
 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Building vue.js app with django api and axios #5883 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7822245/Building.vue.js.app.with.django.api.and.axios.5883.Copyleaks.Report.pdf)

**Comment by dentonya** [comment]
> Hello @dentonya,
>  Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
> 
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> 
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> 
> [Building vue.js app with django api and axios #5883 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7822245/Building.vue.js.app.with.django.api.and.axios.5883.Copyleaks.Report.pdf)
> 

Ok...but most of the red flags are from  the code of which the syntax cannot change @Kayere @mercymeave

**Comment by mercymeave** [comment]
Hello @Kayere, the highlights are mostly snippets uniform to all other Javascript and HTML projects. I think we can just proceed with the review.

@dentonya

**Comment by dentonya** [comment]
> Hello @Kayere, the highlights are mostly snippets uniform to all other Javascript and HTML projects. I think we can just proceed with the review.
> 
> @dentonya 

Ook

**Comment by dentonya** [comment]
Final review ??@mercymeave

File: content/articles/build-vue-application-with-django-rest-api-and-axios/index.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+            }
+        }).then(response => this.tasks = response.data);
+    }
+  }
+}
+</script>
+```
+
+Rerunning the project, you should see two sets of tasks. The ones to be done appear on the left while the ponce that is already done should appear on the right. The complete code for this application can be found [here](https://github.com/dentonya/Todo-Application-using-Vue.js-Django-REST-API-and-Axios/tree/master/vue.js/todoapp).
+
```
Comment (mercymeave) [inline content/articles/build-vue-application-with-django-rest-api-and-axios/index.md:332]: Hello @dentonya 

Would you please provide a screenshot showing the successful execution of this part?

Comment (dentonya) [reply content/articles/build-vue-application-with-django-rest-api-and-axios/index.md:332]: ook

Comment (dentonya) [reply content/articles/build-vue-application-with-django-rest-api-and-axios/index.md]: @mercymeave done with changes

**Comment by dentonya** [comment]
@mercymeave done with changes

**Review by mercymeave — APPROVED** [review]
@dentonya 

Great work!

Be sure to check on the changes to improve the quality of the articles.

Moving for the final review.

Thanks.

**Comment by ahmadmardeni1** [comment]
Please upload a proper screenshot. This is ready to be merged. @dentonya 
cc @mercymeave 

Capture

**Comment by dentonya** [comment]
> Please upload a proper screenshot. This is ready to be merged. @dentonya cc @mercymeave
> 
> Capture

@ahmadmardeni1 done with changes

**Comment by ahmadmardeni1** [comment]
Great work team. @dentonya @mercymeave

**Pull request merged**